storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -2248,4 +2248,23 @@ public Object call() {
         }
         return process;
     }
+
+    public static <T> List<T> interleaveAll(List<List<T>> nodeList) {
+        if (nodeList != null && nodeList.size() > 0) {
+            List<T> first = new ArrayList<T>();
+            List<List<T>> rest = new ArrayList<List<T>>();
+            for (List<T> node : nodeList) {
+                if (null != node && node.size() > 0) {
+                  first.add(node.get(0));
+                  rest.add(node.subList(1, node.size()));
```

Comment:
checked that an empty list is returned. So it should be okay as in next recursion step, we ignore empty list.